src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -657,31 +657,43 @@ impl<'tcx> Constructor<'tcx> {
         param_env: ty::ParamEnv<'tcx>,
         other_ctors: &Vec<Constructor<'tcx>>,
     ) -> Vec<Constructor<'tcx>> {
```

Comment:
This has got me thinking... would using `Cow<[Constructor]>` be more efficient if perhaps `Single | ...` and friends are more common?

(If so, we shouldn't make such changes here, but perhaps in some follow-up.)

Reply:
Quite possibly yes. It'd also be not too hard to take `self` instead of `&self`, if we want to avoid cloning. I don't know which would be better